src/ParametersHandler/YarpImplementation/src/YarpImplementation.cpp

```diff
@@ -15,6 +17,73 @@
 
 using namespace BipedalLocomotion::ParametersHandler;
 
+bool YarpImplementation::stringToDuration(const std::string& string, std::chrono::nanoseconds& time)
+{
+    using namespace std::chrono_literals;
+
+    constexpr auto logPrefix = "[YarpImplementation::stringToDuration]";
+    // \\. means that \ is used as escape character for the regexp and again as escape character for
+    // the string literal.
+    const std::regex expression("[0-9]{2}:[0-9]{2}:[0-9]{2}\\.[0-9]{6}\\b");
+    if (!std::regex_match(string, expression))
+    {
+        log()->debug("{} The parameter do not match the excepted expression. Expected: "
+                     "'HH:MM:SS.xxxxxx'. Get: {}.",
```

Comment:
Most of the time we work with seconds. I would suggest accepting also the ``SS.xxxxxxxx`` case.

Reply:
Ok we can support this for the ini file but not in toml since the latest support only the [RFC 3339](https://tools.ietf.org/html/rfc3339) specification

Reply:
Unless we allow casting a normal double into a "time" variable :thinking:

Reply:
As discussed with @S-Dafarra, we decided to support also the conversion from `double` to `std::chrono::nanoseconds` for both the ini https://github.com/ami-iit/bipedal-locomotion-framework/pull/630/commits/e8b59ffd71cb7de875ae10e114dd0a73bfe89a2f and toml https://github.com/ami-iit/bipedal-locomotion-framework/pull/630/commits/67e74a1e44219d36578b93384a915dc49c296787 files